### PR TITLE
[Format] Format yaml files that will be automated

### PR DESCRIPTION
**Pull request by Bobgy**
Part of #3729 

/assign @Ark-kun 
This makes yaml files that need to be automated during release conformed to the tool's auto format.

Also added a script to automate this:
```
./hack/format.sh
```

### CODE REVIEW

File: manifests/kustomize/base/kustomization.yaml [hunk 1]
```diff
@@ -9,103 +8,97 @@ bases:
...
-    newTag: 0.5.1
-  - name: gcr.io/ml-pipeline/metadata-writer
-    newTag: 0.5.1
-  - name: gcr.io/ml-pipeline/cache-server
-    newTag: 0.5.1
-  - name: gcr.io/ml-pipeline/cache-deployer
-    newTag: 0.5.1
-
+- name: gcr.io/ml-pipeline/api-server
+  newTag: 0.5.1
```
Comment (rmgogogo) [inline manifests/kustomize/base/kustomization.yaml:15]: remember you update it to 1.0.0-rc1, is it due to code base sync or due to branch not merge back to master?

Comment (Bobgy) [reply manifests/kustomize/base/kustomization.yaml:15]: As discussed in chat, @Ark-kun prefers merging back the version only after final 1.0 is cut. I think that makes sense, so rcs won't get version bump to master.

**Pull request merged**